x-pack/plugin/build.gradle

```diff
@@ -25,14 +26,13 @@ subprojects {
       project.dependencies.add('featureAwarePlugin', project(':x-pack:test:feature-aware'))
       project.dependencies.add('featureAwarePlugin', project.sourceSets.main.output.getClassesDirs())
 
-      final Task featureAwareTask = project.tasks.create("featureAwareCheck", LoggedExec) {
+      File successMarker = file("$buildDir/markers/featureAware")
+      task featureAwareCheck(type: LoggedExec) {
         description = "Runs FeatureAwareCheck on main classes."
         dependsOn project.configurations.featureAwarePlugin
-
-        final File successMarker = new File(project.buildDir, 'markers/featureAware')
         outputs.file(successMarker)
 
-        executable = new File(project.runtimeJavaHome, 'bin/java')
+        executable = "${project.runtimeJavaHome}/bin/java"
```

Comment:
Let's avoid doing this. I've already removed a bunch of instances of this while performing other refactorings. There is no reason to do these build time checks with a forked JVM. The code being executed here is build logic, not application logic, and therefore there's no benefit to using the runtime JVM yet we pay the cost of forking another JVM. We don't run the compiler, or checkstyle with the runtime Java home, so we shouldn't here either.

Reply:
I completely agree but don't want to pick that up right now, there were so many things I saw that should be cleaned up, but wanted to keep focus on testclusters.

Reply:
Well we are talking about just removing this line not any kind of significant refactor. Let's just go ahead and make this change while we're here.

Reply:
Not quite, it's a exec not a java exec, so there's a bit more to it.

Reply:
Ah, I see.